### PR TITLE
add C-API app for image classification inference

### PR DESCRIPTION
**Pull request by wojtuss**
This patch adds a C-API application for image classification inference using saved models like ResNet50, SE-ResNeXt50, and MoblileNet-v1.

### CODE REVIEW

File: fluid/PaddleCV/image_classification/capi/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,98 @@
+# Copyright (c) 2016 PaddlePaddle Authors. All Rights Reserved.
```
Comment (Sand3r-) [inline fluid/PaddleCV/image_classification/capi/CMakeLists.txt]: 2018*

**Review by Sand3r- — APPROVED** [review]
Great Job @wojtuss 
I've been using this application for testing purposes for quite a while now.
The only thing that could be considered improving pathing issues in parameters.

File: fluid/PaddleCV/image_classification/capi/CMakeLists.txt (hunk 1 above)
Comment (wojtuss) [reply fluid/PaddleCV/image_classification/capi/CMakeLists.txt]: Done.

**Comment by sfraczek** [comment]
After the changes made for this PR it doesn't run for me. 
All I get is:
```
[sfraczek@nervana-skx42 build]$ ./run.sh

--- Used Parameters: -----------------
Inference model: saved_models/MobileNet-v1_baidu
File with list of images: /data/ILSVRC2012/val_list.txt
Directory with images: /data/ILSVRC2012/
Batch size: 50
Padlle num threads: 10
Iterations: 500
Number of batches to skip: 10
Use fake data: 0
Use MKL-DNN: 1
Skip passes: 0
Debug display image: 0
Profile: 1
With labels: 1
One file params: 0
Channels: 3
Resize size: 256
Crop size: 224
--------------------------------------
```
Contents of run.sh:
```
#!/bin/bash
./infer_image_classification \
        --infer_model=saved_models/MobileNet-v1_baidu \
        --data_list=/data/ILSVRC2012/val_list.txt \
        --data_dir=/data/ILSVRC2012/ \
        --batch_size=50 \
        --skip_batch_num=10 \
        --iterations=500  \
        --profile \
        --use_mkldnn \
        --paddle_num_threads=10
```

**Comment by wojtuss** [comment]
I have added various error handling.

**Review by sfraczek — APPROVED** [review]
Now it works. It handled all my errors :)

**Comment by wojtuss** [comment]
@luotao1 , could you please review or comment on this issue?

**Comment by luotao1** [comment]
@wojtuss why not directly use `test_analyzer_resnet50` unit-test?
Or you should move this PR into `paddle/fluid/inference/api/demo_ci`.

**Comment by wojtuss** [comment]
@luotao1 , we would like this app to be independent of the PaddlePaddle build so that we could use it on various PaddlePaddle builds

**Comment by luotao1** [comment]
> we could use it on various PaddlePaddle builds

What're various PaddlePaddle builds do you mean?

I suggest this PR move to `api/demo_ci`, since it could be tested on CI, and ensure its correctness.

Besides, `demo_ci` is independent of the PaddlePaddle build.

File: fluid/PaddleCV/image_classification/capi/stats.cc [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "stats.h"
+#include <algorithm>
+#include <iomanip>
+#include <iostream>
+#include <iostream>
```
Comment (sfraczek) [inline fluid/PaddleCV/image_classification/capi/stats.cc:19]: Duplicated include

**Comment by wojtuss** [comment]
@luotao1 , OK, I will make a PR with the app to the `api/demo_ci` in PaddlePaddle. Thank you!

**Pull request closed without merging**